common/workunit/workunit.cpp

```diff
@@ -13611,6 +13611,34 @@ extern WORKUNIT_API void addTimeStamp(IWorkUnit * wu, StatisticScopeType scopeTy
     wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), scopeType, scopestr, kind, NULL, getTimeStampNowValue(), 1, 0, StatsMergeAppend);
 }
 
+extern WORKUNIT_API void setGraphCost(IWorkUnit * wu, __int64 cost, StatisticScopeType scopeType, const char * scope, StatisticKind kind, unsigned wfid)
+{
+    StringBuffer scopestr;
+    if (wfid && scope && *scope)
+        scopestr.append(WorkflowScopePrefix).append(wfid).append(":").append(scope);
+    else
+        scopestr.set(scope);
+
+    wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), scopeType, scopestr, kind, NULL, cost, 1, 0, StatsMergeReplace);
+}
+
+extern WORKUNIT_API double calculateThorCost(__int64 timeNs, unsigned clusterWidgth)
+{
+    IPropertyTree *costs = queryCostsConfiguration();
+    if (costs)
+    {
+        const char * val;
+
+        val = costs->queryProp("Cost[@name='thor_master']/@value");
```

Comment:
It would make sense to add a getPropReal(IPropertyTree *, xpath, default) to jptree.hpp, or possibly an inline member in the interface.

Reply:
It is worth thinking what it will look like in the yaml file to configure it.
With xpaths thor/@master and thor/@slave you would have

   costs:
     thor:
        master: 10.2
        slave: 12.3

(I think we are going to change our elements to not have a leading upper case letter for compatibility with normal k8s conventions).